Automattic/wp-calypso
# Jetpack Cloud: Add LogRocket

Proof of concept PR for Log Rocket. More details in p1HpG7-rVk-p2.

## Proposed Changes

* Add and register LogRocket as a tracking tool
* Add a `logrocket` feature flag and enable it on Jetpack Cloud Production.
* Add LogRocket user identification for logged-in users, using hashed PII for privacy.
* Initialize LogRocket only on Backup, Server Credentials settings and Activity Log pages.
* Disable all inputs from recording to ensure sensitive data is not being sent to LogRocket. Here is an example of how input fields will look:
  <img src="https://github.com/Automattic/wp-calypso/assets/1488641/379aef6d-0773-4e97-abfa-a6160b3eab70" width="400" />

## Testing Instructions
* Spin up a Jetpack Cloud Live branch and append `?flags=ad-tracking,logrocket` to the URL to enable the feature flags.
* Enable debug logs using the developer console:
  `localStorage.setItem( 'debug', 'calypso:*' );`
* Filter console and network tabs by `logrocket`
* Navigate to the `Backup` or `Settings` page and ensure you see these debug records:
   ![CleanShot 2024-04-08 at 00 04 11@2x](https://github.com/Automattic/wp-calypso/assets/1488641/45cb2bfc-64a6-4029-8a13-83fdab7f88e9)
* Play around checking your backups, downloading some files, updating your server credentials or even change your retention period.
* And check the network tab and there should be some calls to LogRocket.
* After some time, the session recorded should be available in LogRocket dashboard. If you want access to the platform to watch the recording, please ask for access in p1HpG7-rVk-p2.

## Pre-merge Checklist
